Review and fix vulnerability endpoints API integration test

|Component|
|---|
| API Integration Tests |

The vulnerability endpoints API integration test is failing in the test case "Get agent's vulnerabilities (limit and offset)". In this test case, we expect to have vulnerability results from agent 001 and we are getting an empty response.

We should investigate and fix this error. 


Test failing in Wazuh `master` branch.

<details>

<summary>Error output</summary>

```
Format variables:
  protocol:s = 'https'
  host:s = 'localhost'
  port:d = '55000'
  test_login_token = 'eyJhbGciOiJFUzUxMiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJ3YXp1aCIsImF1ZCI6IldhenVoIEFQSSBSRVNUIiwibmJmIjoxNjM3MTkxMzQ2LCJleHAiOjE2MzcxOTIyNDYsInN1YiI6InRlc3RpbmciLCJydW5fYXMiOmZhbHNlLCJyYmFjX3JvbGVzIjpbMV0sInJiYWNfbW9kZSI6IndoaXRlIn0.AVK-fIFdGdwMRfryPQisIL1aMDTzel9k5O8OssEpd3qq6P_i2C4MHcPbEOY7Zv2cCB6Flz8n3IJoBqmMzQU5-05eACySIMrgl6MiJB-wd315eSEtjEwylgbOAvWdPdoqA1sytjN849XO-nj0iYxWoPvDMXQQWxEezlUx-DbBiZ9_tiC3'

Source test stage (line 9):
  - name: Get agent's vulnerabilities (limit and offset)
    request: &vulnerability_request_001
      verify: False
      url: "{protocol:s}://{host:s}:{port:d}/vulnerability/001"
      method: GET
      headers:
        Authorization: "Bearer {test_login_token}"
      params:
        offset: 5
        limit: 1
    response:
      status_code: 200
      json:
        error: 0
        data:
          affected_items:
            - cve: !anystr
              version: !anystr
              architecture: !anystr
              name: !anystr
              type: !anystr
              status: !anystr
              cvss2_score: !anyfloat
              cvss3_score: !anyfloat
              detection_time: !anystr
              severity: !anystr
          failed_items: []
          total_affected_items: !anyint
          total_failed_items: 0
      save:
        json:
          expected_cve: data.affected_items[0].cve
          expected_version: data.affected_items[0].version
          expected_architecture: data.affected_items[0].architecture
          expected_name: data.affected_items[0].name
          expected_type: data.affected_items[0].type
          expected_status: data.affected_items[0].status
          expected_cvss2_score: data.affected_items[0].cvss2_score
          expected_cvss3_score: data.affected_items[0].cvss3_score
          expected_detection_time: data.affected_items[0].detection_time
          expected_severity: data.affected_items[0].severity


Formatted stage:
  name: Get agent's vulnerabilities (limit and offset)
  request:
    headers:
      Authorization: Bearer eyJhbGciOiJFUzUxMiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJ3YXp1aCIsImF1ZCI6IldhenVoIEFQSSBSRVNUIiwibmJmIjoxNjM3MTkxMzQ2LCJleHAiOjE2MzcxOTIyNDYsInN1YiI6InRlc3RpbmciLCJydW5fYXMiOmZhbHNlLCJyYmFjX3JvbGVzIjpbMV0sInJiYWNfbW9kZSI6IndoaXRlIn0.AVK-fIFdGdwMRfryPQisIL1aMDTzel9k5O8OssEpd3qq6P_i2C4MHcPbEOY7Zv2cCB6Flz8n3IJoBqmMzQU5-05eACySIMrgl6MiJB-wd315eSEtjEwylgbOAvWdPdoqA1sytjN849XO-nj0iYxWoPvDMXQQWxEezlUx-DbBiZ9_tiC3
    method: GET
    params:
      limit: 1
      offset: 5
    url: 'https://localhost:55000/vulnerability/001'
    verify: false
  response:
    json:
      data:
        affected_items:
        - architecture: !anystr ''
          cve: !anystr ''
          cvss2_score: !anyfloat ''
          cvss3_score: !anyfloat ''
          detection_time: !anystr ''
          name: !anystr ''
          severity: !anystr ''
          status: !anystr ''
          type: !anystr ''
          version: !anystr ''
        failed_items: []
        total_affected_items: !anyint ''
        total_failed_items: 0
      error: 0
    save:
      json:
        expected_architecture: data.affected_items[0].architecture
        expected_cve: data.affected_items[0].cve
        expected_cvss2_score: data.affected_items[0].cvss2_score
        expected_cvss3_score: data.affected_items[0].cvss3_score
        expected_detection_time: data.affected_items[0].detection_time
        expected_name: data.affected_items[0].name
        expected_severity: data.affected_items[0].severity
        expected_status: data.affected_items[0].status
        expected_type: data.affected_items[0].type
        expected_version: data.affected_items[0].version
    status_code: 200

Errors:
E   tavern.util.exceptions.TestFailError: Test 'Get agent's vulnerabilities (limit and offset)' failed:
    - List item(s) not present in response: [{'cve': <tavern.util.loader.StrSentinel object at 0x7f6b114fa160>, 'version': <tavern.util.loader.StrSentinel object at 0x7f6b114fa580>, 'architecture': <tavern.util.loader.StrSentinel object at 0x7f6b114fa280>, 'name': <tavern.util.loader.StrSentinel object at 0x7f6b114fa1f0>, 'type': <tavern.util.loader.StrSentinel object at 0x7f6b114fa5b0>, 'status': <tavern.util.loader.StrSentinel object at 0x7f6b114fa5e0>, 'cvss2_score': <tavern.util.loader.FloatSentinel object at 0x7f6b114fa610>, 'cvss3_score': <tavern.util.loader.FloatSentinel object at 0x7f6b114fa640>, 'detection_time': <tavern.util.loader.StrSentinel object at 0x7f6b114fa670>, 'severity': <tavern.util.loader.StrSentinel object at 0x7f6b114fa6a0>}]
```

</details>

We should review and fix this API integration test.

## Checks

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [x] Changelog (`CHANGELOG.md`)
<!-- If changes are made to any of the following components, uncomment the corresponding line 
- [x] **Integration tests** without failures for API configuration (`/wazuh-qa/tests/integration/test_api/test_config/`)
- [x] **System tests** for agent enrollment process (`/wazuh-qa/tests/system/test_cluster/test_agent_enrollment`)
- [x] **System tests** for agent info sync process in cluster (`/wazuh-qa/tests/system/test_cluster/test_agent_info_sync`)
- [x] **System tests** for agent key polling (`/wazuh-qa/tests/system/test_cluster/test_agent_key_polling`)
- [x] **System tests** for JWT invalidation (`/wazuh-qa/tests/system/test_jwt_invalidation`)
-->

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

--- a/api/test/integration/conftest.py
+++ b/api/test/integration/conftest.py
@@ -73,7 +73,7 @@ def build_and_up(interval: int = 10, interval_build_env: int = 10, build: bool =
     os.chdir(pwd)
     values = {
         'interval': interval,
-        'max_retries': 60,
+        'max_retries': 90,
         'retries': 0
     }
     values_build_env = {

--- a/api/test/integration/env/base/agent/new.Dockerfile
+++ b/api/test/integration/env/base/agent/new.Dockerfile
@@ -9,4 +9,4 @@ RUN /wazuh/install.sh
 
 COPY base/agent/entrypoint.sh /scripts/entrypoint.sh
 
-HEALTHCHECK --retries=600 --interval=1s --timeout=30s --start-period=30s CMD /usr/bin/python3 /tmp/healthcheck/healthcheck.py || exit 1
+HEALTHCHECK --retries=900 --interval=1s --timeout=30s --start-period=30s CMD /usr/bin/python3 /tmp/healthcheck/healthcheck.py || exit 1

--- a/api/test/integration/env/base/agent/old.Dockerfile
+++ b/api/test/integration/env/base/agent/old.Dockerfile
@@ -2,4 +2,4 @@ FROM public.ecr.aws/o5x5t0j3/amd64/api_development:integration_test_wazuh-agent_
 
 COPY base/agent/entrypoint.sh /scripts/entrypoint.sh
 
-HEALTHCHECK --retries=600 --interval=1s --timeout=30s --start-period=30s CMD /usr/bin/python3 /tmp/healthcheck/legacy_healthcheck.py || exit 1
+HEALTHCHECK --retries=900 --interval=1s --timeout=30s --start-period=30s CMD /usr/bin/python3 /tmp/healthcheck/legacy_healthcheck.py || exit 1

--- a/api/test/integration/env/base/manager/manager.Dockerfile
+++ b/api/test/integration/env/base/manager/manager.Dockerfile
@@ -12,4 +12,4 @@ RUN /wazuh/install.sh
 COPY base/manager/entrypoint.sh /scripts/entrypoint.sh
 
 # HEALTHCHECK
-HEALTHCHECK --retries=600 --interval=1s --timeout=30s --start-period=30s CMD /var/ossec/framework/python/bin/python3 /tmp/healthcheck/healthcheck.py || exit 1
+HEALTHCHECK --retries=900 --interval=1s --timeout=30s --start-period=30s CMD /var/ossec/framework/python/bin/python3 /tmp/healthcheck/healthcheck.py || exit 1

--- a/api/test/integration/env/configurations/vulnerability/manager/configuration_files/nvd_providers.sh
+++ b/api/test/integration/env/configurations/vulnerability/manager/configuration_files/nvd_providers.sh
@@ -1,0 +1,6 @@
+#!/usr/bin/env bash
+
+# Remove providers
+sed -i '/<os>xenial<\/os>/d' /var/ossec/etc/ossec.conf
+sed -i '/<os>trusty<\/os>/d' /var/ossec/etc/ossec.conf
+sed -i '/<os>bionic<\/os>/d' /var/ossec/etc/ossec.conf

--- a/api/test/integration/test_vulnerability_endpoints.tavern.yaml
+++ b/api/test/integration/test_vulnerability_endpoints.tavern.yaml
@@ -3,7 +3,6 @@ test_name: GET /vulnerability/{agent_id}
 
 marks:
   - base_tests
-  - xfail # Review and fix vulnerability endpoints API integration test - https://github.com/wazuh/wazuh/issues/10917
 
 stages:
 

